### PR TITLE
*: fix invalid failpoint caused by typo

### LINKED ISSUE #11734: Some failpoints don't work as expected
There are two fail points that don't work due to typo, it should be changed from `failpoint` to `failpoints`.

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### What is changed and how it works?

Close #11734  

What's Changed: change `failpoint` to `failpoints`.

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by 5kbpers — APPROVED** [review]

File: components/raftstore/src/store/fsm/apply.rs [hunk 1]
```diff
@@ -3445,7 +3445,7 @@ where
             .iter()
             .any(|res| res.region_id == self.delegate.region_id())
             && self.delegate.last_flush_applied_index != applied_index;
-        #[cfg(feature = "failpoint")]
+        #[cfg(feature = "failpoints")]
```
Comment (BusyJay) [inline components/raftstore/src/store/fsm/apply.rs:3448]: How about removing it completely? It should be optimized away in release mode.

Comment (rleungx) [reply components/raftstore/src/store/fsm/apply.rs:3448]: Do you mean remove `apply_on_handle_snapshot_sync` or just the feature?

**Review by NingLin-P — APPROVED** [review]

File: components/raftstore/src/store/fsm/apply.rs (hunk 1 above)
Comment (BusyJay) [reply components/raftstore/src/store/fsm/apply.rs:3448]: Just the feature.

File: components/resolved_ts/src/advance.rs [hunk 2]
```diff
@@ -6,6 +6,8 @@ use std::time::Duration;
 use collections::{HashMap, HashSet};
 use concurrency_manager::ConcurrencyManager;
 use engine_traits::KvEngine;
+#[cfg(feature = "failpoints")]
```
Comment (BusyJay) [inline components/resolved_ts/src/advance.rs:9]: It doesn't need to be guarded.

Comment (rleungx) [reply components/resolved_ts/src/advance.rs:9]: `make clippy` will report a warning.

Comment (BusyJay) [reply components/resolved_ts/src/advance.rs:9]: If two feature cfgs are removed, then there should be no warnings.

File: components/resolved_ts/src/advance.rs [hunk 3]
```diff
@@ -156,8 +157,7 @@ pub async fn region_resolved_ts_store(
     tikv_clients: Arc<Mutex<HashMap<u64, TikvClient>>>,
     min_ts: TimeStamp,
 ) -> Vec<u64> {
-    #[cfg(feature = "failpoint")]
-    (|| fail_point!("before_sync_replica_read_state", |_| regions))();
+    fail_point!("before_sync_replica_read_state", |_| regions.clone());
```
Comment (BusyJay) [inline components/resolved_ts/src/advance.rs:160]: Either the old implementation or the new implementation is wrong. /cc @NingLin-P

**Review by BusyJay — APPROVED** [review]
Rest LGTM

**Pull request merged**